# 01. Importing Libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

# 02. Importing Data

In [7]:
# Import dataframe
df = pd.read_csv(r'C:\Users\Joe\Documents\Data Analytics Course\Instacart Basket Analysis\02 Data\Original Data\orders.csv', index_col = False)

In [9]:
path = r'C:\Users\Joe\Documents\Data Analytics Course\Instacart Basket Analysis'

In [11]:
path

'C:\\Users\\Joe\\Documents\\Data Analytics Course\\Instacart Basket Analysis'

In [13]:
# Use the OS function
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), index_col = False)

In [15]:
# Lookup Dataframe
df.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [17]:
df.tail()

order_id  user_id eval_set  order_number  order_dow  \
3421078   2266710   206209    prior            10          5   
3421079   1854736   206209    prior            11          4   
3421080    626363   206209    prior            12          1   
3421081   2977660   206209    prior            13          1   
3421082    272231   206209    train            14          6   

         order_hour_of_day  days_since_prior_order  
3421078                 18                    29.0  
3421079                 10                    30.0  
3421080                 12                    18.0  
3421081                 12                     7.0  
3421082                 14                    30.0

In [19]:
# See the shape
df.shape

(3421083, 7)

In [23]:
# Info on the dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3421083 entries, 0 to 3421082
Data columns (total 7 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   order_id                int64  
 1   user_id                 int64  
 2   eval_set                object 
 3   order_number            int64  
 4   order_dow               int64  
 5   order_hour_of_day       int64  
 6   days_since_prior_order  float64
dtypes: float64(1), int64(5), object(1)
memory usage: 182.7+ MB


In [29]:
# Checking the columns
df.columns

Index(['order_id', 'user_id', 'eval_set', 'order_number', 'order_dow',
       'order_hour_of_day', 'days_since_prior_order'],
      dtype='object')

In [33]:
# Describe function
df.describe()

order_id       user_id  order_number     order_dow  \
count  3.421083e+06  3.421083e+06  3.421083e+06  3.421083e+06   
mean   1.710542e+06  1.029782e+05  1.715486e+01  2.776219e+00   
std    9.875817e+05  5.953372e+04  1.773316e+01  2.046829e+00   
min    1.000000e+00  1.000000e+00  1.000000e+00  0.000000e+00   
25%    8.552715e+05  5.139400e+04  5.000000e+00  1.000000e+00   
50%    1.710542e+06  1.026890e+05  1.100000e+01  3.000000e+00   
75%    2.565812e+06  1.543850e+05  2.300000e+01  5.000000e+00   
max    3.421083e+06  2.062090e+05  1.000000e+02  6.000000e+00   

       order_hour_of_day  days_since_prior_order  
count       3.421083e+06            3.214874e+06  
mean        1.345202e+01            1.111484e+01  
std         4.226088e+00            9.206737e+00  
min         0.000000e+00            0.000000e+00  
25%         1.000000e+01            4.000000e+00  
50%         1.300000e+01            7.000000e+00  
75%         1.600000e+01            1.500000e+01  
max         2.300000e+01            3.000000e+01

In [35]:
# Using Lists
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), nrows=1000)

In [37]:
# Creating a list
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']

In [39]:
# Import with only columns from the list
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), usecols = vars_list)

In [41]:
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

# Import Products CSV

In [47]:
df_prods = pd.read_csv(os.path.join(path,'02 Data', 'Original Data', 'products.csv'), index_col = False)

In [49]:
# First 20 rows and last 35 rows
df_prods.head(20), df_prods.tail(35)

(    product_id                                       product_name  aisle_id  \
 0            1                         Chocolate Sandwich Cookies        61   
 1            2                                   All-Seasons Salt       104   
 2            3               Robust Golden Unsweetened Oolong Tea        94   
 3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
 4            5                          Green Chile Anytime Sauce         5   
 5            6                                       Dry Nose Oil        11   
 6            7                     Pure Coconut Water With Orange        98   
 7            8                  Cut Russet Potatoes Steam N' Mash       116   
 8            9                  Light Strawberry Blueberry Yogurt       120   
 9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
 10          11                                  Peach Mango Juice        31   
 11          12                         

In [51]:
# Names of the cloumns
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [53]:
# Number of rows and columns
df_prods.shape

(49693, 5)

# Final Questions

In [55]:
# Find the maximum value of the 'aisle_id' column
df_prods['aisle_id'].max()

134

The maximum value of the aisle column is 134.

In [59]:
# What's the data type of the 'department_id' column?
df_prods['department_id'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 49693 entries, 0 to 49692
Series name: department_id
Non-Null Count  Dtype
--------------  -----
49693 non-null  int64
dtypes: int64(1)
memory usage: 388.4 KB


The data type for the department_id column is int64.